### PR TITLE
add setuptools-scm to template

### PR DESCRIPTION
**Pull request by stevebachmeier**
### Description

- *Category*:  other
- *JIRA issue*: https://jira.ihme.washington.edu/browse/MIC-4243

### Changes and notes
 
this adds setuptools-scm to the repo template. The primary
motivation here is to get the dynamic versions of model repos
(i.e. version \<latest tag\> \<distance to tag\> \<working directory state\>)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -105,3 +105,6 @@ venv.bak/
 
 # pycharm
 .idea/
+
+# Version file
+src/*/_version.py
```
Comment (mattkappel) [inline .gitignore:110]: Why not this? 

```suggestion
src/{{cookiecutter.package_name}}/_version.py
```

**Review by mattkappel — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (mattkappel) [reply .gitignore:110]: Nevermind, this is the gitignore for the repo

**Review by rmudambi — APPROVED** [review]
Did you already do this in a model repo? I must have missed the PR if so

**Comment by mattkappel** [comment]
Oh maybe add _version to the cookiecutter package name's gitignore

**Review by albrja — APPROVED** [review]

**Comment by stevebachmeier** [comment]
> Did you already do this in a model repo? I must have missed the PR if so

Yes https://github.com/ihmeuw/vivarium_nih_us_cvd/pull/171

**Comment by stevebachmeier** [comment]
> Oh maybe add _version to the cookiecutter package name's gitignore

@mattkappel Oh good catch - I added to the wrong gitignore!

**Pull request merged**